src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,52 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
-        $seen = [];
+        $senderAliases = [];
+
+        if (\PHP_VERSION_ID >= 80000) {
+            $senderAliases = $this->getSendersFromAttributes($envelope);
+        }
 
         foreach (HandlersLocator::listTypes($envelope) as $type) {
             foreach ($this->sendersMap[$type] ?? [] as $senderAlias) {
-                if (!\in_array($senderAlias, $seen, true)) {
-                    if (!$this->sendersLocator->has($senderAlias)) {
-                        throw new RuntimeException(sprintf('Invalid senders configuration: sender "%s" is not in the senders locator.', $senderAlias));
-                    }
-
-                    $seen[] = $senderAlias;
-                    $sender = $this->sendersLocator->get($senderAlias);
-                    yield $senderAlias => $sender;
-                }
+                $senderAliases[] = $senderAlias;
```

Comment:
previous FIFO, now LIFO

Reply:
Wow, I didn't even think about it. Yes, array_unique should keep the same order.